src/curvefit/solvers/solvers.py

```diff
@@ -99,4 +100,32 @@ def fit(self, data, x_init=None, options=None):
                 xs_opt.append(self.solver.x_opt)
             
             self.x_opt = xs_opt[np.argmin(fun_vals)]
-            self.fun_val_opt = np.min(fun_vals)
\ No newline at end of file
+            self.fun_val_opt = np.min(fun_vals)
+
+
+class SmartInitialization(Composite):
```

Comment:
How would we run a `SmartInitialization` model and also a `GaussianMixtureIntegration`? I'm a little confused about the inheritance that happens with the `Composite`

Reply:
do you mean putting smart init inside gaussian mixtures? it would be something like 
```
smart_init = SmartInitialization(update_prior_fun)
gm = GaussianMixtures(...)
gmi = GaussianMixtures(gm)
model = CoreModel(...)
gmi.set_solver(smart_init)
gmi.set_model_instance(model)
gmi.fit(data)
```